php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1305 @@
+    InternalSyntaxToken eosToken
+    ) {
+    return ClassPropertyDeclarationTreeImpl.constant(constToken, separatedList(firstDeclaration, additionalDeclarations), eosToken);
+  }
+  
+  public ClassPropertyDeclarationTree classVariableDeclaration(
+    List<SyntaxToken> modifierTokens, 
+    VariableDeclarationTree firstVariable,
+    Optional<List<Tuple<InternalSyntaxToken, VariableDeclarationTree>>> additionalVariables, 
+    InternalSyntaxToken eosToken
+    ) {
+    return ClassPropertyDeclarationTreeImpl.variable(modifierTokens, separatedList(firstVariable, additionalVariables), eosToken);
+  }
+
+  public MethodDeclarationTree methodDeclaration(
+    Optional<List<SyntaxToken>> modifiers, 
+    InternalSyntaxToken functionToken, 
+    Optional<InternalSyntaxToken> referenceToken,
+    IdentifierTree name, 
+    ParameterListTree parameters, 
+    Tree body
+    ) {
+    return new MethodDeclarationTreeImpl(optionalList(modifiers), functionToken, referenceToken.orNull(), name, parameters, body);
+  }
+  
+  public FunctionDeclarationTree functionDeclaration(
+    InternalSyntaxToken functionToken, 
+    Optional<InternalSyntaxToken> referenceToken, 
+    IdentifierTree name, 
+    ParameterListTree parameters,
+    BlockTree body
+    ) {
+    return new FunctionDeclarationTreeImpl(functionToken, referenceToken.orNull(), name, parameters, body);
+  }
+  
+  public ParameterListTree parameterList(
+    InternalSyntaxToken leftParenthesis,
+    Optional<Tuple<ParameterTree, Optional<List<Tuple<InternalSyntaxToken, ParameterTree>>>>> parameters,
+    InternalSyntaxToken rightParenthesis
+    ) {
+    SeparatedList<ParameterTree> separatedList = SeparatedList.empty();
+    if (parameters.isPresent()) {
+      separatedList = separatedList(parameters.get().first(), parameters.get().second());
+    }
+    return new ParameterListTreeImpl(leftParenthesis, separatedList, rightParenthesis);
+  }
+  
+  public ParameterTree parameter(
+    Optional<Tree> classType, 
+    Optional<InternalSyntaxToken> ampersand, 
+    Optional<InternalSyntaxToken> ellipsis, 
+    InternalSyntaxToken identifier,
+    Optional<Tuple<InternalSyntaxToken, ExpressionTree>> eqAndInitValue
+    ){
+    InternalSyntaxToken eqToken = null;
+    ExpressionTree initValue = null;
+    if (eqAndInitValue.isPresent()) {
+      eqToken = eqAndInitValue.get().first();
+      initValue = eqAndInitValue.get().second();
+    }
+    VariableIdentifierTree varIdentifier = new VariableIdentifierTreeImpl(new IdentifierTreeImpl(identifier));
+    return new ParameterTreeImpl(classType.orNull(), ampersand.orNull(), ellipsis.orNull(), varIdentifier, eqToken, initValue);
+  }
+  
+  public SeparatedList<NamespaceNameTree> interfaceList(NamespaceNameTree first, Optional<List<Tuple<InternalSyntaxToken, NamespaceNameTree>>> others) {
+    return separatedList(first, others);
+  }
+  
+  // FIXME (PY) Do Traits belong to statements or declarations?
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)